linq4j/src/main/java/org/apache/calcite/linq4j/EnumerableDefaults.java

```diff
@@ -1091,38 +1125,44 @@ public boolean moveNext() {
                 return true;
               }
               if (!outers.moveNext()) {
-                if (unmatchedKeys != null) {
-                  // We've seen everything else. If we are doing a RIGHT or FULL
-                  // join (leftNull = true) there are any keys which right but
-                  // not the left.
-                  List<TInner> list = new ArrayList<>();
-                  for (TKey key : unmatchedKeys) {
-                    for (TInner tInner : innerLookup.get(key)) {
-                      list.add(tInner);
-                    }
-                  }
-                  inners = Linq4j.enumerator(list);
+                if (innersUnmatched != null) {
+                  inners = Linq4j.enumerator(new ArrayList<>(innersUnmatched));
```

Comment:
Do we really need a new list here? Couldn't this just be:
`inners = Linq4j.enumerator(innersUnmatched);`
?